plugins/base/src/main/kotlin/translators/documentables/DescriptionSections.kt

```diff
@@ -0,0 +1,326 @@
+        sourceSets = availableSourceSets
+    ) {
+        availableSourceSets.forEach { sourceSet ->
+            inheritors[sourceSet]?.forEach { classlike: DRI ->
+                inheritorRow(classlike, logger, sourceSet)
+            }
+        }
+    }
+}
+
+private fun PageContentBuilder.DocumentableContentBuilder.sharedSourceSetOnlyInheritorsSectionContent(
+    inheritors: Map<DokkaConfiguration.DokkaSourceSet, List<DRI>>,
+    logger: DokkaLogger,
+) {
+    val uniqueInheritors = inheritors.values.flatten().toSet()
+    tableSectionContentBlock(
+        blockName = "Inheritors",
+        kind = ContentKind.Inheritors,
+    ) {
+        uniqueInheritors.forEach { classlike ->
+            inheritorRow(classlike, logger)
+        }
+    }
+}
+
+private fun PageContentBuilder.DocumentableContentBuilder.tableSectionContentBlock(
+    blockName: String,
+    kind: ContentKind,
+    sourceSets: Set<DokkaConfiguration.DokkaSourceSet> = mainSourcesetData,
+    body: PageContentBuilder.TableBuilder.() -> Unit,
+) {
+    header(KDOC_TAG_HEADER_LEVEL, text = blockName, kind = kind, sourceSets = sourceSets)
+    table(
+        kind = kind,
+        sourceSets = sourceSets,
+        extra = mainExtra + SimpleAttr.header(blockName)
+    ) {
+        body()
+    }
+}
+
+private fun PageContentBuilder.TableBuilder.inheritorRow(
+    classlike: DRI, logger: DokkaLogger, sourceSet: DokkaConfiguration.DokkaSourceSet? = null,
+) = row {
+    link(
+        text = classlike.friendlyClassName()
+            ?: classlike.toString().also { logger.warn("No class name found for DRI $classlike") },
+        address = classlike,
+        sourceSets = sourceSet?.let { setOf(it) } ?: mainSourcesetData
+    )
+}
+
+private fun WithScope.inheritors() = safeAs<WithExtraProperties<Documentable>>()
+    ?.let { it.extra[InheritorsInfo] }
+    ?.let { inheritors -> inheritors.value.filter { it.value.isNotEmpty() } }
+    .orEmpty()
+
+private fun DRI.friendlyClassName() = classNames?.substringAfterLast(".")
+
+private fun TagWrapper.isNotEmpty() = this.children.isNotEmpty()
+
+private fun <V> Map<DokkaConfiguration.DokkaSourceSet, V>.fallback(sourceSets: List<DokkaConfiguration.DokkaSourceSet>): V? =
+    sourceSets.firstOrNull { it in this.keys }.let { this[it] }
+
+/**
+ * Used for multi-value tags (e.g. params) when values are missed on some platforms.
+ * It this case description is inherited from parent platform.
+ * E.g. if param hasn't description in JVM, the description is taken from common.
+ */
+private fun Set<DokkaConfiguration.DokkaSourceSet>.getPossibleFallback(sourceSet: DokkaConfiguration.DokkaSourceSet) =
+    this.filter { it.sourceSetID in sourceSet.dependentSourceSets }
+
+private val specialTags: Set<KClass<out TagWrapper>> =
+    setOf(Property::class, Description::class, Constructor::class, Param::class, See::class)
+
+private fun <T> Map<String, SourceSetDependent<T>>.availableSourceSets() = values.flatMap { it.keys }.toSet()
```

Comment:
Related to the whole section of different functions at the end of the file. I understand that it migrated from `DefaultPageCreator.kt` where it's written this way, but now could be a good time to change it while we still can:

I'm personally a big believer in ordering functions in such a way that the code can be read from top to bottom without having to jump up and down the file or even without too much scrolling, where one function kind of flows into another (if at all possible). It's a common approach, but I can't google a nice article about it :(

A simple example would be the following:
```
fun main() {
    a()
    b()
    c()
}

fun a() {
    // a
}

fun b() {
    b1()
    b2()
}

fun b1() {
    // b1
}

fun b2() {
    // b2
}

fun c() {
    c1()
}

fun c1() {
    // c1
}
```

Alternatively, you could do
```
a()
b()
c()

b1()
b2()
c1()
```

but, if functions were long, that would make you jump up and down more than if it was sequential and in the order of being called and closer to the place where it was called.

I went to see if Android Kotlin style guides had anything to say on this (it covers more details and cases than the official guide), and funnily enough they covered [this case exactly](https://developer.android.com/kotlin/style-guide#top-level_declarations):

> For example, new functions are not just habitually added to the end of the file, as that would yield “chronological by date added” ordering, which is not a logical ordering.

___

I see that you've done that for some functions already, so I'm proposing to move these functions out of here as well - closer to where they are called.

For instance, `fallback` function can go from line 310 to 159. `inheritors` from 301 to 219, `isNotEmpty` from 308 to 186, and so on.

As for the `specialTags` property, while I found no convention on the order of top-level properties, I would put it at the very top as that's where you naturally expect to find properties within a class.